File: Daily Test/C++/Matrix - Count Empty Cells 23.6.2025.cpp
#include <iostream>
#include <unordered_set>
using namespace std;

int main() {
    int r, c, q;
    cin >> r >> c >> q;

    unordered_set<int> rows, cols;

    while (q--) {
        int x, y;
        cin >> x >> y;
        rows.insert(x);
        cols.insert(y);
        int filled = rows.size() * c + cols.size() * r - rows.size() * cols.size();
        cout << r * c - filled << endl;
    }

    return 0;
}

File: Daily Test/Python/Matrix - Count Empty Cells 23.6.2025.py
r, c = map(int, input().split())
q = int(input())

rows = set()
cols = set()

for _ in range(q):
    x, y = map(int, input().split())
    rows.add(x)
    cols.add(y)
    filled = len(rows) * c + len(cols) * r - len(rows) * len(cols)
    print(r * c - filled)